### PR TITLE
Test Runner and Examples of Use

### PR DESCRIPTION
**Pull request by jrans**
## THIS PR

Create test runner to go through all configurable rules and make sure they are all configured as we desire.

We can pass either `null` to show we have not included the rule, or an object of the form 
```
{
  fail: ['failString']
  pass: ['passString', read('example-rule.pass')]
}
```
with ability to read in a js file if placed in the correct folder i.e.  `test/fixtures/example-rule.pass.js

Resolves: #188
Resolves: #189

### CODE REVIEW

File: eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  rules: {
+    "space-infix-ops": ["error", {"int32Hint": false}],
```
Comment (nelsonic) [inline eslintrc.js]: @jrans can we add a comment with a link to the issue where the rule is discussed on each line of the config file? e.g: 
```js
"space-infix-ops": ["error", {"int32Hint": false}], // https://github.com/dwyl/goodparts/issues/32
"max-len": ["error", 80], // https://github.com/dwyl/goodparts/issues/186
```

File: test/fixtures/max-len.fail.js [hunk 2]
```diff
@@ -0,0 +1 @@
+var line = 'waaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaayTooLong';
```
Comment (nelsonic) [inline test/fixtures/max-len.fail.js]: 👍

**Comment by nelsonic** [comment]
@jrans looks good. left comment regarding including comments in the config file... thoughts?

File: eslintrc.js (hunk 1 above)
Comment (jrans) [inline eslintrc.js]: Thanks will ensure this going forward.

**Comment by nelsonic** [comment]
👀 👍 :shipit:

**Pull request merged**